Makefile.dep

```diff
@@ -791,9 +794,7 @@ ifneq (,$(filter random,$(USEMODULE)))
     USEMODULE += hashes
   endif
 
-  ifeq (,$(filter puf_sram,$(USEMODULE)))
-    FEATURES_OPTIONAL += periph_hwrng
-  endif
+  FEATURES_OPTIONAL += $(call find_first_of_these_provided_features, puf_sram periph_hwrng)
```

Comment:
I think one would prefer `periph_hwrng` over `puf_sram` (at least currently), as `periph_hwrng` has less impact on boot up time and (at least in case of warm boots) more entropy.

Reply:
Agreed

Reply:
What I was doing here resembles the `FEATURES_REQUIRED_ONE_OUT_OF` but `FEATURES_OPTIONAL_ONE_OUT_OF`, I think I should probably re-work it that way.

Reply:
Can I then just invert the ordering here?

Reply:
Yep :-)